### PR TITLE
UI 2.0 Updates

### PR DESCRIPTION
**Pull request by moustafaelsisy**
This PR takes care of applying a dark theme with green accents to our web app. It also enahcnes the font set used, and opts for a more minimal design to the application
* Added the `Encode Sans Condensed` font to the app (https://fonts.google.com/specimen/Encode+Sans+Condensed)

* Added a css file dedicated to the chat page to be loaded along with the main styles

* Updated the styling rules used in the application to achieve the new theme look

* Rearranged the navbar greeting for logged in users to the right side

screenshot from 2018-06-29 00-21-14
screenshot from 2018-06-29 00-21-19
screenshot from 2018-06-29 00-21-21
screenshot from 2018-06-29 00-21-23
screenshot from 2018-06-29 00-21-25

### CODE REVIEW

File: src/main/webapp/WEB-INF/tags/base.tag [hunk 1]
```diff
@@ -22,6 +22,7 @@
 <head>
   <title><%= title != null ? title : "EastCode Chat" %></title>
   <link rel="stylesheet" href="/css/main.css">
+  <link href="https://fonts.googleapis.com/css?family=Encode+Sans+Condensed" rel="stylesheet">
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/tags/base.tag:25]: This adds the new font family as a possible font to be used by the app

File: src/main/webapp/WEB-INF/tags/base.tag [hunk 2]
```diff
@@ -31,12 +32,13 @@
...
-      <a>Hello <%= request.getSession().getAttribute("user") %>!</a>
-    <% } else{ %>
-      <a href="/login">Login</a>
+    <% if(request.getSession().getAttribute("user") == null){ %>
+          <a href="/login">Login</a>
     <% } %>
     <a href="/about.jsp">About</a>
+    <% if(request.getSession().getAttribute("user") != null){ %>
+      <a id="greeting">Hello <%= request.getSession().getAttribute("user") %>!</a>
+    <% } %>
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/tags/base.tag:41]: This allows us to have the greeting separately. Note the addition of the `id` attribute to the "Hello username" anchor

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 3]
```diff
@@ -22,14 +22,7 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
     <jsp:attribute name="title"><%= conversation.getTitle() %></jsp:attribute>
     <jsp:attribute name="headContent">
-         <style>
-            #chat {
-              background-color: white;
-              height: 500px;
-              overflow-y: scroll
-            }
-          </style>
-
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/view/chat.jsp:32]: These have been moved along with the new styles to `/css/chat.css`

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 4]
```diff
@@ -22,14 +22,7 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
+            <link rel="stylesheet" href="/css/chat.css">
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/view/chat.jsp:25]: This loads the new css file for the chat page

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 5]
```diff
@@ -41,7 +34,7 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
     </jsp:attribute>
     <jsp:attribute name="bodyContent">
             <h1><%= request.getAttribute("customChatTitle") != null ? request.getAttribute("customChatTitle") : conversation.getTitle() %>
-              <a href="" style="float: right">&#8635;</a></h1>
+              <a href="" id="refresh">&#8635;</a></h1>
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/view/chat.jsp:37]: I am applying this styling through this new id in the chat page's new stylesheet

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 6]
```diff
@@ -59,18 +52,16 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
+                <% if (request.getSession().getAttribute("user") != null) { %>
+                <form action='<%= (String)request.getAttribute("messagePostUrl") %>' method="POST">
+                    <input id="message-input" type="text" name="message">
+                    <br/>
+                    <button id="message-submit" type="submit">Send</button>
+                </form>
+                <% } else { %>
+                  <p><a href="/login">Login</a> to send a message.</p>
+                <% } %>
+            </div>
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/view/chat.jsp:65]: Just wrapped the same piece of code within the div tag of id `input-area`

Comment (moustafaelsisy) [reply src/main/webapp/WEB-INF/view/chat.jsp:65]: And added IDs to the input box and submit button

File: src/main/webapp/WEB-INF/view/conversations.jsp [hunk 7]
```diff
@@ -27,15 +27,15 @@
...
+              <div class="form-group" style="margin-bottom: 16px;">
                 <label class="form-control-label">Title:</label>
               <input type="text" name="conversationTitle">
             </div>
 
             <button type="submit">Create</button>
           </form>
 
-          <hr/>
+          <hr class="groove"/>
```
Comment (moustafaelsisy) [inline src/main/webapp/WEB-INF/view/conversations.jsp:38]: Again, the styling for the `groove` class is applied in the `main.css` stylesheet

File: src/main/webapp/css/chat.css [hunk 8]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (moustafaelsisy) [inline src/main/webapp/css/chat.css:1]: The styles here will apply only on the chat page c:

File: src/main/webapp/css/main.css [hunk 9]
```diff
@@ -14,45 +14,83 @@
  */
```
Comment (moustafaelsisy) [inline src/main/webapp/css/main.css:14]: The styles here apply on all pages, along with any specific css files that get loaded only on given pages

**Review by ivannpe — APPROVED** [review]

File: src/main/webapp/WEB-INF/tags/base.tag (hunk 1 above)
Comment (blessinga98) [reply src/main/webapp/WEB-INF/tags/base.tag:25]: Love this font!

File: src/main/webapp/WEB-INF/tags/base.tag (hunk 2 above)
Comment (blessinga98) [reply src/main/webapp/WEB-INF/tags/base.tag:41]: This makes a lot of sense.

**Review by blessinga98 — APPROVED** [review]

**Pull request merged**